### ISSUE TITLE
when I finish black-box testing, I get this error

### ISSUE THREAD

**Issue by jiaxian-lin** [EMResearch/EvoMaster#440]
```
* Going to save 108 tests to src/em
* [31m[ERROR] [0m[33mEvoMaster process terminated abruptly. This is likely a bug in EvoMaster. Please copy&paste the following stacktrace, and create a new issue on [34mhttps://github.com/EMResearch/EvoMaster/issues[0m[0m
com.google.gson.JsonSyntaxException: com.google.gson.stream.MalformedJsonException: Use JsonReader.setLenient(true) to accept malformed JSON at line 1 column 107 path $
        at com.google.gson.Gson.assertFullConsumption(Gson.java:908)
        at com.google.gson.Gson.fromJson(Gson.java:898)
        at com.google.gson.Gson.fromJson(Gson.java:846)
        at com.google.gson.Gson.fromJson(Gson.java:817)
        at org.evomaster.core.output.clustering.metrics.DistanceMetricErrorText.calculateDistance(DistanceMetricErrorText.kt:35)
        at org.evomaster.core.output.clustering.metrics.DistanceMetricErrorText.calculateDistance(DistanceMetricErrorText.kt:21)
        at org.evomaster.core.output.clustering.DBSCANClusterer.getNeighbours(DBSCANClusterer.kt:78)
        at org.evomaster.core.output.clustering.DBSCANClusterer.performCLustering(DBSCANClusterer.kt:111)
        at org.evomaster.core.output.Clusterer.cluster(Clusterer.kt:67)
        at org.evomaster.core.output.TestSuiteSplitter.conductClustering(TestSuiteSplitter.kt:94)
        at org.evomaster.core.output.TestSuiteSplitter.split(TestSuiteSplitter.kt:50)
        at org.evomaster.core.Main$Companion.writeTests(Main.kt:504)
        at org.evomaster.core.Main$Companion.writeTests$default(Main.kt:486)
        at org.evomaster.core.Main$Companion.initAndRun(Main.kt:160)
        at org.evomaster.core.Main$Companion.main(Main.kt:71)
        at org.evomaster.core.Main.main(Main.kt)
Caused by: com.google.gson.stream.MalformedJsonException: Use JsonReader.setLenient(true) to accept malformed JSON at line 1 column 107 path $
        at com.google.gson.stream.JsonReader.syntaxError(JsonReader.java:1562)
        at com.google.gson.stream.JsonReader.checkLenient(JsonReader.java:1403)
        at com.google.gson.stream.JsonReader.doPeek(JsonReader.java:541)
        at com.google.gson.stream.JsonReader.peek(JsonReader.java:424)
        at com.google.gson.Gson.assertFullConsumption(Gson.java:904)
        ... 15 common frames omitted
```

by the way, how can I know the API coverage after black-box testing

**Comment by arcuri82** [comment]
hi.
 thx for reporting this bug. Until it is fixed, try to run with:

`--testSuiteSplitType NONE` 

as that should deactivate the code with the bug.

btw, what do you mean exactly with "API coverage"?  can you provide some more clarifications?

**Comment by jiaxian-lin** [comment]
> hi. thx for reporting this bug. Until it is fixed, try to run with:
> 
> `--testSuiteSplitType NONE`
> 
> as that should deactivate the code with the bug.
> 
> btw, what do you mean exactly with "API coverage"? can you provide some more clarifications?

for example, the API spec has 5 unique API, and after testing, evomaster can run successfully 4 API, one left just can receive 404, then the API coverage is 80%

**Comment by arcuri82** [comment]
hi,
 thx. all endpoints should be covered. if one gets 404, it is still called, isn't it? so that would still be 100% coverage.
or do you mean coverage when getting a response with 2xx? in that case, we cannot guarantee it, as it depends on business logic of the system

however, another possibility is if the OpenApi/Swagger schema has issues, and the path for that endpoint is wrong (in that case, it will never be called)

**Comment by jiaxian-lin** [comment]
thx for your reply, this means that we can use API coverage to evaluate evomaster's ability to generate valid requests and a more intuitive understanding of the effect of this test. 
just a suggestion

**Comment by arcuri82** [comment]
thx. so, in the generated tests, you would like to see something like a comment (maybe at the beginning of the generated file) stating something like `It was possible to generate successful requests with status code 2xx for 4 out of 5 (80%) endpoints`?

**Comment by arcuri82** [comment]
or would you prefer it to have in on the console logs? or both?

**Comment by jiaxian-lin** [comment]
I think console logs may be better. 👍

**Comment by arcuri82** [comment]
This issue should now be fixed in the new 1.4.0 release
